lua/telekasten.lua

```diff
@@ -183,7 +183,7 @@ local function generate_note_filename(uuid, title)
     end
 end
 
-local function check_dir_and_ask(dir, purpose)
+local function check_dir_and_ask(dir, purpose, callback)
```

Comment:
this is the biggest source of change. Added `callback` param to `check_dir_and_ask`, now all callsites to `check_dir_and_ask` need to pass the "after selection" operations as a callback.